3scale/nginx.conf

```diff
@@ -51,7 +51,7 @@ http {
             proxy_http_version 1.1;
             proxy_buffering off;
             proxy_set_header Upgrade $http_upgrade;
-            proxy_set_header Connection "upgrade";
+            proxy_set_header Connection $http_connection;
```

Comment:
I'm not actually sure how 3scale behaves. Unfortunately I cannot test it as the config has been broken since Friday, see https://issues.redhat.com/browse/RHCLOUD-21158 . I hope it will be fixed soon, then I'll see what it does. (and also try to deploy this stuff on k8s). If 3scale would always add "Connection: upgrade", that'd be a bummer indeed  -- but it would also be rather weird to do that, TBH. The Connection: header is hop-by-hop, yes, but a proxy should still respect what the browser actually asked for, not blindly declare every HTTP request as an upgrade request. In that sense, our 3scale emulation feels buggy -- the question is just if the official one is as well :wink:

Reply:
Oh! I found some raw header dumps in https://issues.redhat.com/browse/COCKPIT-865 -- so official 3scale indeed does *not* set `Connection: upgrade` on each request. *Phew*! But I'll rework the commit message to describe this better.

Reply:
updated commit message